database/database_functions.py

```diff
@@ -1,64 +1,96 @@
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
-from database_models import Persons, Rooms, Allocations, BASE
+from database.database_models import Persons, Rooms, BASE
 
-ENGINE = create_engine('sqlite:///amity.db')
 
-BASE.metadata.create_all(ENGINE)
-DBSESSION = sessionmaker(bind=ENGINE)
-SESSIONS = DBSESSION()
 
-def get_persons():
-    persons = SESSIONS.query(Persons).all()
-    p_dct = {}
-    for person in persons:
-        p_dct.update({person.p_name:person.p_type})
+class AmityDatabase(object):
 
-    return p_dct
+    '''
+    class contains function for the database
+    '''
 
-def get_rooms():
-    rooms = SESSIONS.query(Rooms).all()
-    r_dct = {}
-    for room in rooms:
-        r_dct.update({room.r_name:room.r_type})
+    def __init__(self):
+        pass
 
-    return r_dct
+    def initialize(self, filename):
+        '''
+        The function initaializes the database through a new databse or an existing database.
+        '''
+        self.ENGINE = create_engine('sqlite:///'+filename)
+        BASE.metadata.create_all(self.ENGINE)
+        self.DBSESSION = sessionmaker(bind=self.ENGINE)
+        self.SESSIONS = self.DBSESSION()
 
-def get_allocations():
-    allocations = SESSIONS.query(Allocations).all()
-    al_dct = {}
-    for allocation in allocations:
-        al_dct.update({allocation.p_name:[allocation.o_name, allocation.ls_name]})
+    def get_persons(self, filename):
+        '''
+        The function get the person from database and add the person to a dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        persons = self.SESSIONS.query(Persons).all()
+        person_dictionary = {}
+        for person in persons:
+            person_dictionary.update({person.person_name:person.person_type})
+        return person_dictionary
 
-    return al_dct
+    def get_rooms(self, filename):
+        '''
+        The function get the rooms from database and add the rooms to the dictionary with regards
+        to room name and room type to a dictionary
+        '''
+        # calling the initialize function to use the specified the database
+        self.initialize(filename)
+        rooms = self.SESSIONS.query(Rooms).all()
+        room_dictionary = {}
+        for room in rooms:
+            room_dictionary.update({room.room_name:room.room_type})
+        return room_dictionary
 
+    def get_allocations(self, filename):
+        '''
+        The function get the allocations from database and adds the allocations to the dictionary
+        with regards to person name and person type and the room they belong to into a dictionary.
+        '''
+        # calling the initialize function to use the specified the database
```

Comment:
This comment is not really necessary, because there is a docstring explaining what the function does